test/support/builders/atom_watcher_event.js

```diff
@@ -0,0 +1,76 @@
+/* @flow */
+
+const metadata = require('../../../core/metadata')
+const events = require('../../../core/local/steps/event')
+
+/*::
+import type { Stats } from 'fs'
+import type { AtomWatcherEvent, Action, Kind, Batch } from '../../../core/local/steps/event'
+*/
+
+function randomPick /*:: <T> */ (elements /*: Array<T> */) /*: T */{
+  const l = elements.length
+  const i = Math.floor(Math.random() * l)
+  return elements[i]
+}
+
+module.exports = class AtomWatcherEventBuilder {
+  /*::
+  _event: AtomWatcherEvent
+  */
+
+  constructor () {
+    this._event = {
+      action: randomPick(events.ACTIONS),
+      kind: randomPick(events.KINDS),
+      path: '/',
+      _id: '/'
+    }
+  }
+
+  build () /*: AtomWatcherEvent */ {
+    return this._event
+  }
+
+  action (newAction /*: Action */) /*: this */ {
+    if (!events.ACTIONS.includes(newAction)) {
+      throw new Error(`Unknown event action "${newAction}". Available actions: ${events.ACTIONS.join(', ')}.`)
```

Comment:
Aren't we doing flow's job twice?

Reply:
I didn't have Flow's checks in place when I wrote those methods so it might be overkill. I left them because I often run the linter only right before I push the commits (or even after sometimes…) and thought it could be useful to have this check.

I can remove them.